### PR TITLE
fix: add syncing state to trade loading

### PR DESCRIPTION
**Pull request by ianlapham**

### CODE REVIEW

File: src/lib/hooks/routing/clientSideSmartOrderRouter.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export async function getClientSideQuote(
   )
 }
 
+/**  Used to keep quotes up to date given a certain block age. Returns undefined if past limit. */
 export function useFreshQuote(quoteResult: GetQuoteResult | undefined, maxBlockAge = 10): GetQuoteResult | undefined {
```
Comment (zzmp) [inline src/lib/hooks/routing/clientSideSmartOrderRouter.ts:104]: nit: This is a filter function, so you could also rename it something like `useFilterFreshQuote` or `useFilterQuote`.
Probably for another PR, though.

File: src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts [hunk 2]
```diff
@@ -78,26 +79,39 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
       return { error: true }
     }
   }, [config, params, queryArgs, wrapType])
+
```
Comment (zzmp) [inline src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts:82]: nit: why all these new newlines?

File: src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts [hunk 3]
```diff
@@ -106,9 +120,13 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
 
     // Returns the last trade state while syncing/loading to avoid jank from clearing the last trade while loading.
     if (!error) {
+      if (isStale) {
+        return { state: TradeState.LOADING, trade: undefined }
+      }
```
Comment (zzmp) [inline src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts:125]: This should be linked with isDebouncing with an `else if`, as they're all mutually exclusive.

**Review by zzmp — APPROVED** [review]
lgtm, just some formatting nits.

File: src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts [hunk 4]
```diff
@@ -133,5 +151,5 @@ export default function useClientSideSmartOrderRouterTrade<TTradeType extends Tr
       return { state: TradeState.VALID, trade }
     }
     return { state: TradeState.INVALID, trade: undefined }
-  }, [currencyIn, currencyOut, quoteResult, error, route, queryArgs, trade, isDebouncing, tradeType])
+  }, [currencyIn, currencyOut, error, quoteResult, route, queryArgs, trade, isDebouncing, isLoading, tradeType])
```
Comment (zzmp) [inline src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts:154]: Include `isStale` in the list of deps.

File: src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts (hunk 2 above)
Comment (ianlapham) [reply src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts:82]: I inentionally made this one because i thought it made things more readable

**Pull request merged**